editor/src/components/canvas/ui-jsx-canvas.tsx

```diff
@@ -1291,7 +1314,7 @@ function renderJSXElement(
     elementIsIntrinsic && !elementIsBaseHTML ? filterDataProps(elementProps) : elementProps
 
   if (FinalElement == null) {
-    return createMissingElement(jsx)
+    return throwForMissingElement(jsx, imports, topLevelNames)
```

Comment:
what would be the error message if we wouldn't handle this case and call createElement with a `null` or `undefined`? would there be an error, or just a silent missing component?

Reply:
Let me try!

Reply:
@balazsbajorics Oddly it seems to be really variable, testing around editing a file and reloading the editor I've seen the debug version of the undefined error that React normally gives, the production version of that with the minified error URL and a `ReferenceError`.

Reply:
hah, that's... not the best. the undefined error (if reliable) would be the best, but I don't think it's worth more of your time to try to chase it now

Reply:
Sorry for the late review here...

In looking at this, I'm wondering if we're better off removing this custom handling and replacing the line below with 
```if (FinalElement != null && TP.containsPath(templatePath, validPaths)) {```
Doing that would lead to us trying to use the standard React handling of invalid components, which gives an error:
```Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined.```
Though granted the "Check the render method of " part of that error message will be wrong because it is inside the canvas's custom rendering (it should read "Check the render method of App"):

![Screenshot_20201009_173047](https://user-images.githubusercontent.com/1044774/95608608-47eacc00-0a55-11eb-80b6-a1a167fb37fc.png)

Reply:
I'm not quite sure what you're describing position-wise, but if it's effectively similar to @balazsbajorics's suggestion I noticed that it wasn't very consistent in behaviour when it came to reloads and changes.

Reply:
Hmmm I've just been playing with this and I'm not seeing that weird inconsistent behaviour. OK I'll play about with this a bit more and come back to you because I'd much rather we could rely on the default React behaviour than have custom behaviour if possible.

Reply:
I'm all for that, entirely possible that a change to the logic makes it work more effectively. It _sounds_ like your logic makes this function just return an `undefined` instead of invoking `React.createElement` with `undefined.